src/main/java/games/strategy/triplea/ui/ResourceBar.java

```diff
@@ -64,15 +64,17 @@ public void gameDataChanged(final Change change) {
     try {
       final PlayerID player = gameData.getSequence().getStep().getPlayerId();
       if (player != null) {
-        for (int i = 0; i < resources.size(); i++) {
-          final String quantity = resources.get(i).getFormatter().format(resources.get(i).getValue(player, gameData));
+        for (int i = 0; i < resourceStats.size(); i++) {
+          final IStat resourceStat = resourceStats.get(i);
```

Comment:
Might as well extract a new local for the `JLabel` while you're doing this for `resourceStat` to avoid the repeated calls to `labels.get(i)` below.

```java
final JLabel label = labels.get(i);
```

Reply:
Updated.